### PR TITLE
Component screenshot with preliminary rendering

### PR DESCRIPTION
**Pull request by mkfl3x**

### CODE REVIEW

File: remote-robot/src/main/kotlin/com/intellij/remoterobot/fixtures/Fixture.kt [hunk 1]
```diff
@@ -63,11 +63,14 @@ abstract class Fixture(
...
-     * Use ImageIO.write method with "png" formatName
+     * Get component screenshot.
+     * Use ImageIO.write() method with "png" formatName
+     *
+     * @param withPainting false by default. Set true for rendering the component before capturing screenshot
+     * @return BufferedImage specified as .png
      */
-    fun getScreenshot(): BufferedImage {
-        val bytes = remoteRobot.ideRobotClient.makeScreenshot(remoteComponent.id)
+    fun getScreenshot(withPainting: Boolean = false): BufferedImage {
```
Comment (nizienko) [inline remote-robot/src/main/kotlin/com/intellij/remoterobot/fixtures/Fixture.kt:72]: Could you please check how the method is usable from the java code.

File: robot-server-plugin/src/main/kotlin/com/intellij/remoterobot/RobotServerImpl.kt [hunk 2]
```diff
@@ -259,7 +259,11 @@ class RobotServerImpl {
                 get("/{componentId}/screenshot") {
                     call.dataResultRequest({
                         val componentId = call.parameters["componentId"] ?: throw IllegalArgumentException("empty componentId")
-                        ideRobot.makeScreenshot(componentId)
+                        val paintingMode = call.parameters["withPainting"].toBoolean()
```
Comment (nizienko) [inline robot-server-plugin/src/main/kotlin/com/intellij/remoterobot/RobotServerImpl.kt:262]: I would name the property `isPaintingMode`

**Review by nizienko — CHANGES_REQUESTED** [review]

File: remote-robot/src/main/kotlin/com/intellij/remoterobot/fixtures/Fixture.kt (hunk 1 above)
Comment (mkfl3x) [reply remote-robot/src/main/kotlin/com/intellij/remoterobot/fixtures/Fixture.kt:72]: it's definitely usable

File: robot-server-plugin/src/main/kotlin/com/intellij/remoterobot/RobotServerImpl.kt (hunk 2 above)
Comment (mkfl3x) [reply robot-server-plugin/src/main/kotlin/com/intellij/remoterobot/RobotServerImpl.kt]: renamed

**Pull request merged**